corehq/apps/ota/views.py

```diff
@@ -61,15 +68,23 @@ def restore(request, domain, app_id=None):
     tags = [
         u'status_code:{}'.format(response.status_code),
     ]
+    env = settings.SERVER_ENVIRONMENT
+    if (env, domain) in settings.RESTORE_TIMING_DOMAINS:
+        tags.append(u'domain:{}'.format(domain))
     datadog_counter('commcare.restores.count', tags=tags)
     if timing_context is not None:
+        datadog_histogram(
+            'commcare.restores.total_time',
+            timing_context.duration,
+            tags=tags,
+        )
         for timer in timing_context.to_list(exclude_root=True):
-            # Only record leaf nodes so we can sum to get the total
-            if timer.is_leaf_node:
+            if timer.name in TIMED_RESTORE_SEGMENTS:
+                segment = TIMED_RESTORE_SEGMENTS[timer.name]
                 datadog_histogram(
```

Comment:
Already discussed this with @snopoke 

After looking at the existing data in datadog for a long time, I'm beginning to wonder if the histogram metric type is very useful here, given the aggregation interval is so short (10s, see [What is the histogram metric type?](https://help.datadoghq.com/hc/en-us/articles/205638045-What-is-the-histogram-metric-type-)). Quote from [datadog metric types docs](https://help.datadoghq.com/hc/en-us/articles/206955236-Metric-types-in-Datadog): `histogram( ... )` "Should be called multiple times during an agent check (otherwise you have no distribution)."

Are we sending enough data to get a meaningful distribution? The avg, 95% and max stats are closer together than I would expect. Maybe the many segments previously tracked were causing a problem?